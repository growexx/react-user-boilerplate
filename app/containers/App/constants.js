/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

import {
  FAV_ICON_152_152,
  FAV_ICON_16_16,
  FAV_ICON_32_32,
  APPLE_TOUCH_ICON,
  FAV_ICON_192_192,
} from '../../images/favicons';
export const LOAD_REPOS = 'boilerplate/App/LOAD_REPOS';
export const LOAD_REPOS_SUCCESS = 'boilerplate/App/LOAD_REPOS_SUCCESS';
export const LOAD_REPOS_ERROR = 'boilerplate/App/LOAD_REPOS_ERROR';
export const LOAD_APP = 'boilerplate/App/LOAD_APP';
export const FAV_ICONS = [
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '512x512',
    href: FAV_ICON_152_152,
  },

  {
    rel: 'icon',
    type: 'image/png',
    sizes: '192x192',
    href: FAV_ICON_192_192,
  },
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '32x32',
    href: FAV_ICON_32_32,
  },
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '16x16',
    href: FAV_ICON_16_16,
  },
  {
    rel: 'apple-touch-icon',
    href: APPLE_TOUCH_ICON,
  },
];
