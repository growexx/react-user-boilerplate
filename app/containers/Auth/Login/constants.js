export const CHANGE_EMAIL = 'boilerplate/Login/CHANGE_EMAIL';
export const CHANGE_PASSWORD = 'boilerplate/Login/CHANGE_PASSWORD';
export const ERROR = 'boilerplate/Login/ERROR';
export const LOADING = 'boilerplate/Login/LOADING';
export const SUCCESS = 'boilerplate/Login/SUCCESS';
export const LOGIN = 'boilerplate/Login/LOGIN';
export const RESET = 'boilerplate/Login/RESET';
export const GOOGLE_LOGIN = 'boilerplate/Login/GOOGLELOGIN';
export const FACEBOOK_LOGIN = 'boilerplate/Login/FACEBOOKLOGIN';
