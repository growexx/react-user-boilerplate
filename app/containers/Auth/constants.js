export const AUTH_TYPE = ['LOGIN', 'REGISTER'];

// API-ROUTES
export const API_URL = 'https://b0257f3aa104.ngrok.io';
export const AUTH = 'auth';
export const API_ENDPOINTS = {
  LOGIN: `${API_URL}/${AUTH}/signin`,
  LIST: 'https://randomuser.me/api/?results=5&inc=name,gender,email,nat&noinfo',
  LIST_AVATAR:
    'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png',
  EXPORT_CSV: '',
};
