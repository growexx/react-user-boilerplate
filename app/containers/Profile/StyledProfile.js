import styled from 'styled-components';

export const StyledProfile = styled.div`
  display: flex;
  flex-direction: column;
  .ant-card {
    flex: 1 !important;
    cursor: default;
  }
`;

export const CardExtraContainer = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  p {
    color: @primary-color;
    font-size: @font-size-base;
    margin-bottom: 0;
  }
`;
