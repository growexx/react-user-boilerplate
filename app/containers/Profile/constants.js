/*
 *
 * Profile constants
 *
 */
export const PROFILE_PLACEHOLDER =
  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum";

export const DATA_TEST_IDS = {
  ABOUT_EDIT: 'About_Edit',
  ABOUT_SAVE: 'About_Save',
  ABOUT_EDITOR: 'About_Editor',
  EDUCATION_EDIT: 'Education_Edit',
  EDUCATION_SAVE: 'Education_Save',
  EXPERIENCE_EDIT: 'Experience_Edit',
  EXPERIENCE_SAVE: 'Experience_Save',
  LICENSEANDCERTIFICATION_EDIT: 'LicensesAndCertifications_edit',
  LICENSEANDCERTIFICATION_SAVE: 'LicensesAndCertifications_save',
};
