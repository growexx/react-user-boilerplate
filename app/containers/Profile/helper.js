/* eslint-disable consistent-return */
/**
 * options for draft-js-export-html
 * for more options: https://github.com/sstur/draft-js-utils/blob/master/packages/draft-js-export-html/README.md
 */
export const options = {
  inlineStyles: {
    // Override default element (`strong`).
    BOLD: { element: 'b' },
  },
  defaultBlockTag: 'div',
  inlineStyleFn: styles => {
    const key = 'color-';
    const color = styles.filter(value => value.startsWith(key)).first();

    if (color) {
      return {
        element: 'span',
        style: {
          color: color.replace(key, ''),
        },
      };
    }
  },
  blockStyleFn: block => {
    if (block.getData().get('color')) {
      return {
        style: {
          color: block.getData().get('color'),
        },
      };
    }
  },
};
