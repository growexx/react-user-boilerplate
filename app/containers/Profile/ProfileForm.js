/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/anchor-is-valid */
/**
 *
 * ProfileForm Form
 *
 */

import React from 'react';
import { EditorState } from 'draft-js';
import { FormattedHTMLMessage } from 'react-intl';
import ReactHtmlParser from 'react-html-parser';
import { Helmet } from 'react-helmet';
import { Card, Button } from 'antd';
import { stateToHTML } from 'draft-js-export-html';
import FileUpload from 'components/FileUpload/Loadable';
import InlineEdit from 'components/InlineEdit';
import { CardExtraContainer } from './StyledProfile';
import { DATA_TEST_IDS, PROFILE_PLACEHOLDER } from './constants';
import { options } from './helper';
import messages from './messages';
import RichTextEditor from '../../components/RichTextEditor';

class ProfileForm extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      nameContent: 'John Doe',
      designationContent: 'Software Engineer at GrowExx',
      locationContent: 'Ahmedabad, India',
      aboutContent: EditorState.createEmpty(),
      experienceContent: EditorState.createEmpty(),
      educationContent: EditorState.createEmpty(),
      licensesAndCertificationsContent: EditorState.createEmpty(),
      editAbout: false,
      editExperience: false,
      editEducation: false,
      editLicensesAndCertifications: false,
    };
  }

  isEditModeOn = state => this.state[state] === true;

  isContentEdited = state =>
    state.getCurrentContent().getPlainText('\u0001').length > 0
      ? stateToHTML(state.getCurrentContent(), options)
      : PROFILE_PLACEHOLDER;

  render() {
    const {
      nameContent,
      designationContent,
      locationContent,
      aboutContent,
      educationContent,
      experienceContent,
      licensesAndCertificationsContent,
    } = this.state;
    return (
      <div>
        <Helmet>
          <title>ProfileForm</title>
          <meta name="description" content="Description of ProfileForm" />
        </Helmet>
        <div className="u-mb-1">
          <InlineEdit
            value={nameContent}
            onSave={value => this.setState({ nameContent: value })}
            placeholder="Enter Name"
          />
        </div>
        <div className="u-mb-1">
          <InlineEdit
            value={designationContent}
            onSave={value => this.setState({ designationContent: value })}
            placeholder="Enter Designation"
          />
        </div>
        <div className="u-mb-5">
          <InlineEdit
            value={locationContent}
            onSave={value => this.setState({ locationContent: value })}
            placeholder="Enter Location"
          />
        </div>
        <Card
          hoverable
          type="inner"
          title="About"
          extra={
            <CardExtraContainer>
              {this.isEditModeOn('editAbout') ? (
                <Button
                  data-testid={DATA_TEST_IDS.ABOUT_SAVE}
                  onClick={() => {
                    this.setState({ editAbout: false });
                  }}
                >
                  <FormattedHTMLMessage {...messages.save} />
                </Button>
              ) : (
                <Button
                  data-testid={DATA_TEST_IDS.ABOUT_EDIT}
                  onClick={() => {
                    this.setState({ editAbout: true });
                  }}
                >
                  <FormattedHTMLMessage {...messages.edit} />
                </Button>
              )}
            </CardExtraContainer>
          }
        >
          {this.isEditModeOn('editAbout') ? (
            <RichTextEditor
              testId={DATA_TEST_IDS.ABOUT_EDITOR}
              value={aboutContent}
              onChange={value => {
                this.setState({
                  aboutContent: value,
                });
              }}
            />
          ) : (
            ReactHtmlParser(`${this.isContentEdited(aboutContent)}`)
          )}
        </Card>
        <br />
        <Card
          hoverable
          type="inner"
          title="Experience"
          extra={
            <CardExtraContainer>
              {this.isEditModeOn('editExperience') ? (
                <Button
                  data-testid={DATA_TEST_IDS.EXPERIENCE_SAVE}
                  onClick={() => {
                    this.setState({ editExperience: false });
                  }}
                >
                  <FormattedHTMLMessage {...messages.save} />
                </Button>
              ) : (
                <Button
                  data-testid={DATA_TEST_IDS.EXPERIENCE_EDIT}
                  onClick={() => {
                    this.setState({ editExperience: true });
                  }}
                >
                  <FormattedHTMLMessage {...messages.edit} />
                </Button>
              )}
            </CardExtraContainer>
          }
        >
          {this.isEditModeOn('editExperience') ? (
            <RichTextEditor
              value={experienceContent}
              onChange={value => {
                this.setState({
                  experienceContent: value,
                });
              }}
            />
          ) : (
            ReactHtmlParser(`${this.isContentEdited(experienceContent)}`)
          )}
        </Card>
        <br />
        <Card
          hoverable
          type="inner"
          title="Education"
          extra={
            <CardExtraContainer>
              {this.isEditModeOn('editEducation') ? (
                <Button
                  data-testid={DATA_TEST_IDS.EDUCATION_SAVE}
                  onClick={() => {
                    this.setState({ editEducation: false });
                  }}
                >
                  <FormattedHTMLMessage {...messages.save} />
                </Button>
              ) : (
                <Button
                  data-testid={DATA_TEST_IDS.EDUCATION_EDIT}
                  onClick={() => {
                    this.setState({ editEducation: true });
                  }}
                >
                  <FormattedHTMLMessage {...messages.edit} />
                </Button>
              )}
            </CardExtraContainer>
          }
        >
          {this.isEditModeOn('editEducation') ? (
            <RichTextEditor
              value={educationContent}
              onChange={value => {
                this.setState({
                  educationContent: value,
                });
              }}
            />
          ) : (
            ReactHtmlParser(`${this.isContentEdited(educationContent)}`)
          )}
        </Card>
        <br />
        <Card
          hoverable
          type="inner"
          title="Licenses and Certificates"
          extra={
            <CardExtraContainer>
              {this.isEditModeOn('editLicensesAndCertifications') ? (
                <Button
                  data-testid={DATA_TEST_IDS.LICENSEANDCERTIFICATION_SAVE}
                  onClick={() => {
                    this.setState({ editLicensesAndCertifications: false });
                  }}
                >
                  <FormattedHTMLMessage {...messages.save} />
                </Button>
              ) : (
                <Button
                  data-testid={DATA_TEST_IDS.LICENSEANDCERTIFICATION_EDIT}
                  onClick={() => {
                    this.setState({ editLicensesAndCertifications: true });
                  }}
                >
                  <FormattedHTMLMessage {...messages.edit} />
                </Button>
              )}
            </CardExtraContainer>
          }
        >
          {this.isEditModeOn('editLicensesAndCertifications') ? (
            <RichTextEditor
              value={licensesAndCertificationsContent}
              onChange={value => {
                this.setState({
                  licensesAndCertificationsContent: value,
                });
              }}
            />
          ) : (
            ReactHtmlParser(
              `${this.isContentEdited(licensesAndCertificationsContent)}`,
            )
          )}
        </Card>
        <br />
        <Card hoverable type="inner" title="Upload Resume">
          <FileUpload />
        </Card>
      </div>
    );
  }
}

export default ProfileForm;
