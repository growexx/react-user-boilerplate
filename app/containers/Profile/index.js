/**
 *
 * Profile
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import { Card } from 'antd';
import ImageUpload from 'components/ImageUpload/Loadable';
import { StyledProfile } from './StyledProfile';
import ProfileForm from './ProfileForm';

export function Profile() {
  return (
    <div>
      <Helmet>
        <title>Profile</title>
        <meta name="description" content="Description of Profile" />
      </Helmet>
      <Card>
        <StyledProfile>
          <ImageUpload />
          <ProfileForm />
        </StyledProfile>
      </Card>
    </div>
  );
}

export default Profile;
