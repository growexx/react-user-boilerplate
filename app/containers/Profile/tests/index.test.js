/**
 *
 * Tests for Profile
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { fireEvent, render } from 'react-testing-library';
/**
 * user-event is a companion library for Testing Library
 * that provides more advanced simulation of browser
 * interactions than the built-in fireEvent method
 * https://testing-library.com/docs/ecosystem-user-event/
 */
import userEvent from '@testing-library/user-event';
/**
 * This import adds some helpful assertions.
 * Custom jest matchers to test the state of the DOM
 * https://github.com/testing-library/jest-dom
 */
import 'jest-dom/extend-expect';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { browserHistory } from 'react-router-dom';
import { TEST_IDS } from 'components/InlineEdit/stub';
import history from 'utils/history';
import Profile from '../index';
import configureStore from '../../../configureStore';
import { DATA_TEST_IDS } from '../constants';
jest.mock('draft-js/lib/generateRandomKey', () => jest.fn(() => '123'));

let store;

const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <Profile />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<Profile />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('Should click Save Buttons', () => {
    const { getByTestId } = componentWrapper();
    fireEvent.click(getByTestId(DATA_TEST_IDS.ABOUT_EDIT));
    fireEvent.click(getByTestId(DATA_TEST_IDS.ABOUT_SAVE));
    expect(getByTestId(DATA_TEST_IDS.ABOUT_EDIT)).toBeTruthy();
    fireEvent.click(getByTestId(DATA_TEST_IDS.EXPERIENCE_EDIT));
    fireEvent.click(getByTestId(DATA_TEST_IDS.EXPERIENCE_SAVE));
    expect(getByTestId(DATA_TEST_IDS.EXPERIENCE_EDIT)).toBeTruthy();
    fireEvent.click(getByTestId(DATA_TEST_IDS.EDUCATION_EDIT));
    fireEvent.click(getByTestId(DATA_TEST_IDS.EDUCATION_SAVE));
    expect(getByTestId(DATA_TEST_IDS.EDUCATION_EDIT)).toBeTruthy();
    fireEvent.click(getByTestId(DATA_TEST_IDS.LICENSEANDCERTIFICATION_EDIT));
    fireEvent.click(getByTestId(DATA_TEST_IDS.LICENSEANDCERTIFICATION_SAVE));
    expect(
      getByTestId(DATA_TEST_IDS.LICENSEANDCERTIFICATION_EDIT),
    ).toBeTruthy();
  });
  it('Should double click on name and write in input box', async () => {
    const { getAllByTestId } = componentWrapper();
    // show the input
    userEvent.dblClick(getAllByTestId(TEST_IDS.INPUT_VALUE)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_EDIT)[0]).toBeInTheDocument();
    // write on the input
    fireEvent.change(getAllByTestId(TEST_IDS.INPUT_EDIT)[0], {
      target: {
        value: 'testInputChanged',
      },
    });
    // save the input
    fireEvent.click(getAllByTestId(TEST_IDS.SAVE_BUTTON)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_VALUE)[0]).toBeInTheDocument();
  });
  it('Should double click on designation and write in input box', async () => {
    const { getAllByTestId } = componentWrapper();
    // show the input
    userEvent.dblClick(getAllByTestId(TEST_IDS.INPUT_VALUE)[1]);
    expect(getAllByTestId(TEST_IDS.INPUT_EDIT)[0]).toBeInTheDocument();
    // write on the input
    fireEvent.change(getAllByTestId(TEST_IDS.INPUT_EDIT)[0], {
      target: {
        value: 'testInputChanged',
      },
    });
    // save the input
    fireEvent.click(getAllByTestId(TEST_IDS.SAVE_BUTTON)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_VALUE)[1]).toBeInTheDocument();
  });
  it('Should double click on location and write in input box', async () => {
    const { getAllByTestId } = componentWrapper();
    // show the input
    userEvent.dblClick(getAllByTestId(TEST_IDS.INPUT_VALUE)[2]);
    expect(getAllByTestId(TEST_IDS.INPUT_EDIT)[0]).toBeInTheDocument();
    // write on the input
    fireEvent.change(getAllByTestId(TEST_IDS.INPUT_EDIT)[0], {
      target: {
        value: 'testInputChanged',
      },
    });
    // save the input
    fireEvent.click(getAllByTestId(TEST_IDS.SAVE_BUTTON)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_VALUE)[2]).toBeInTheDocument();
  });
  it('Should fire mouse down on other target and input should be disappeared', async () => {
    const { getAllByTestId } = componentWrapper();
    // show the input
    userEvent.dblClick(getAllByTestId(TEST_IDS.INPUT_VALUE)[2]);
    expect(getAllByTestId(TEST_IDS.INPUT_EDIT)[0]).toBeInTheDocument();
    fireEvent.mouseDown(getAllByTestId(TEST_IDS.INPUT_VALUE)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_VALUE)[2]).toBeInTheDocument();
  });
});
