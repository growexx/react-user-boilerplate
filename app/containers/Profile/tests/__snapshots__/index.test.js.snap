// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> Should render and match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .ant-card {
  -webkit-flex: 1 !important;
  -ms-flex: 1 !important;
  flex: 1 !important;
  cursor: default;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 p {
  color: #4d186e;
  font-size: 14px;
  margin-bottom: 0;
}

.c1 .inputValue {
  height: auto;
  padding: 8px;
}

.c1 .inputValue:hover {
  cursor: text;
  background: rgba(9,30,66,0.04);
}

.c1 .placeHolderView {
  color: #808080 !important;
}

<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="c0"
      >
        <div>
          <div
            class="u-mb-1"
          >
            <div
              class="c1"
            >
              <div
                class="inputValue"
                data-testid="INPUT_VALUE"
              >
                John Doe
              </div>
            </div>
          </div>
          <div
            class="u-mb-1"
          >
            <div
              class="c1"
            >
              <div
                class="inputValue"
                data-testid="INPUT_VALUE"
              >
                Software Engineer at GrowExx
              </div>
            </div>
          </div>
          <div
            class="u-mb-5"
          >
            <div
              class="c1"
            >
              <div
                class="inputValue"
                data-testid="INPUT_VALUE"
              >
                Ahmedabad, India
              </div>
            </div>
          </div>
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  About
                </div>
                <div
                  class="ant-card-extra"
                >
                  <div
                    class="c2"
                  >
                    <button
                      class="ant-btn"
                      data-testid="About_Edit"
                      type="button"
                    >
                      <span>
                        Edit
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
            </div>
          </div>
          <br />
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  Experience
                </div>
                <div
                  class="ant-card-extra"
                >
                  <div
                    class="c2"
                  >
                    <button
                      class="ant-btn"
                      data-testid="Experience_Edit"
                      type="button"
                    >
                      <span>
                        Edit
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
            </div>
          </div>
          <br />
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  Education
                </div>
                <div
                  class="ant-card-extra"
                >
                  <div
                    class="c2"
                  >
                    <button
                      class="ant-btn"
                      data-testid="Education_Edit"
                      type="button"
                    >
                      <span>
                        Edit
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
            </div>
          </div>
          <br />
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  Licenses and Certificates
                </div>
                <div
                  class="ant-card-extra"
                >
                  <div
                    class="c2"
                  >
                    <button
                      class="ant-btn"
                      data-testid="LicensesAndCertifications_edit"
                      type="button"
                    >
                      <span>
                        Edit
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
            </div>
          </div>
          <br />
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  Upload Resume
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
